src/sandstorm/web-session.capnp

```diff
@@ -76,7 +76,17 @@ interface WebSession @0xa50711a14d35a8ce extends(Grain.UiSession) {
   # `clientStream` is the capability which will receive server -> client messages, while
   # serverStream represents client -> server.
 
-  # TODO(someday): Add WebDAV and CalDAV methods?
+  # WebDAV methods
+  propfind @7 (path :Text, content :PropfindContent, context :Context) -> Response;
+  proppatch @8 (path :Text, content :ProppatchContent, context :Context) -> Response;
+  mkcol @9 (path :Text, content :MkcolContent, context :Context) -> Response;
+  copy @10 (path :Text, context :Context) -> Response;
+  move @11 (path :Text, context :Context) -> Response;
+  lock @12 (path :Text, content :LockContent, context :Context) -> Response;
+  unlock @13 (path :Text, content :UnlockContent, context :Context) -> Response;
+  acl @14 (path :Text, content :AclContent, context :Context) -> Response;
+  report @15 (path :Text, content :ReportContent, context :Context) -> Response;
+  options @16 (path :Text, context :Context) -> Response;
```

Comment:
I'm somewhat concerned that passing through OPTIONS could have security problems independent of DAV. OPTIONS is a general HTTP method that has been overloaded for a lot of uses. As an example, we certainly wouldn't want to allow an app to respond to an OPTIONS request with `Access-Control-*`; Sandstorm has strong opinions on what those headers should contain. Of course, we don't pass through those headers, so that's not a problem. I am worried, though, that there might be bad things you can do with OPTIONS that could slip through; it's hard to say since it's so overloaded.

Can we narrow down the precise ways in which OPTIONS is used by WebDAV apps and support them more directly? It looks like it's only for the "DAV" header? Perhaps we could instead have an `isDav()` method for this?